### DISCUSSION TITLE
how to access mediarecorder / stream

### DISCUSSION THREAD

**Discussion by cipherhell** [samhirtarif/react-audio-recorder#41] [Q&A] [answered]
how to access the stream to e.g. display volume? Would you consider exporting that?

**Comment by samhirtarif** [comment]
@cipherhell - With the current implementation of the package, the media recorder cannot be accessed.

That said, it can be made available fairly easily with a not-so-extensive change. But I had originally left it out because I didn't feel like the end user would have a need for it. Could you maybe explain in more detail why you need it? When you say "display volume" do you mean to say that you want to visualise the volume of whatever is currently being fed to the recorder as a waveform?

**Comment by cipherhell** [comment]
hey, yeah, no I wanted to integrate it with hark and visualize a volumebar. I'm unsure how to patch it or pull request because noob

**Comment by samhirtarif** [comment]
Are you referring to this package? https://www.npmjs.com/package/hark

And no worries, I'll make the relevant changes today and drop an example here of how you can use it.
And just an FYI, I am working on visualizing the audio input as the recorder gets input. This would hopefully be released coming weekend.

**Comment by cipherhell** [comment]
no way, that's insane, thanks, yes I was referring to that package. It's gotten more popular I think, and it's quite convenient to use to get speech onset events.

Can I ask, is there any way to set the contraints of the mediarecorder in your library? 
Mediarecorder seems limited by default as far as I can see, cannot get chrome to record 320kbps 48khz audio, even if constrains can be set

**Comment by samhirtarif** [comment]
Are you referring to the `audioBitsPerSecond` option in the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder/MediaRecorder#parameters)? There's currently no way to set that, but definitely something I can add.

And the docs do state that the default bit rate is 2.5Mbps.
<img width="757" alt="image" src="https://github.com/samhirtarif/react-audio-recorder/assets/89391183/81cfd7f9-da5a-4364-b9b4-fc068cbe9614">

**Comment by cipherhell** [comment]
yes that's what I was basically referring to, but if you read closely, the 2.5Mbps applies to video, the audio is 128kbps

**Comment by samhirtarif** [comment, marked as answer]
@cipherhell - I've published changes that expose media recorder now.
You can find an example of the changes [here](https://stackblitz.com/edit/react-ts-5vjuch?file=App.tsx). Please mark this comment as the answer and close the discussion if this solves your problems.

**Comment by samhirtarif** [comment]
@cipherhell - If your problem was resolved, please do mark the above comment as the answer and close the discussion. Thanks.

**Reply by cipherhell** [reply]
thanks for the elaboration!
